kuzzleio/kuzzle
# Allow to set request error for request:onError pipes

## What does this PR do ?

This PR allows to manually set another error than the original one when using a pipe on `request:onError` event.  
Before this PR, the only way to change the error returned to the client was to instantiate a new request: 
```
      const
        {
          data,
          options
        } = request.serialize(),
        customRequest = this.context.constructors.Request(data, options);

      customRequest.setError(new this.context.errors.UnauthorizedError('WRONG_CREDENTIALS'));

      return customRequest;
```

### How should this be manually tested?

  - Step 1 : In a plugin, add the following hook on `request:onError`
```
  async postLoginError (request) {
    if (request.input.controller === 'auth' && request.input.action === 'login' && request.status === 401) {
      const
        {
          data,
          options
        } = request.serialize(),
        customRequest = this.context.constructors.Request(data, options);

      customRequest.setError(new this.context.errors.UnauthorizedError('WRONG_CREDENTIALS'));

      return customRequest;
    }

    return request;
  }
```
  - Step 2 : Then try to login into Kuzzle with wrong credentials, you should see a 401 error with `WRONG_CREDENTIALS`
